src/protocol/v6/structures/Vector.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Bolt\protocol\v6\structures;
+
+use Bolt\packstream\Bytes;
+use Bolt\protocol\IStructure;
+
+/**
+ * Class Vector
+ * Immutable
+ *
+ * @author Michal Stefanak
+ * @link https://github.com/neo4j-php/Bolt
+ * @link https://www.neo4j.com/docs/bolt/current/bolt/structure-semantics/#structure-vector
+ * @package Bolt\protocol\v6\structures
+ */
+class Vector implements IStructure
+{
+    public function __construct(
+        public readonly Bytes $type_marker,
+        public readonly Bytes $data
+    ) {
+    }
+
+    public function __toString(): string
+    {
+        return json_encode([(string)$this->type_marker, (string)$this->data]);
+    }
+
+    private static array $formats = ['s', 'l', 'q'];
+
+    /**
+     * Encode array as vector structure
+     * @param int[]|float[] $data
+     * @return self
+     * @throws \InvalidArgumentException
+     */
+    public static function encode(array $data): self
+    {
+        if (count($data) === 0) {
+            throw new \InvalidArgumentException('Vector cannot be empty');
+        }
+        if (count($data) > 4096) {
+            throw new \InvalidArgumentException('Vector cannot have more than 4096 elements');
+        }
```

Comment:
This is up to you, but I'd like to bring to your attention that the official drivers deliberately do not check the vector dimension before sending it. Reasons:
 * The size is not limited by the bolt protocol specification. Any size is valid according to bolt.
 * The server might decide at any point to broaden the permitted dimension values.
 * Other vendors besides Neo4j that implement the Bolt protocol are free to choose their own limits (or none).

Reply:
I didn't try what will happen if the array is bigger than allowed limit. I was following documentation at https://neo4j.com/docs/cypher-manual/25/values-and-types/vector/#valid-values

So you are suggesting not to force these limits at driver level and just let application implementing the driver to receive neo4j error?